client/ayon_core/hosts/houdini/plugins/publish/collect_files_for_cleaning_up.py

```diff
@@ -0,0 +1,85 @@
+import pyblish.api
+import os
+from ayon_core.pipeline import AYONPyblishPluginMixin
+from ayon_core.hosts.houdini.api import lib
+
+
+class CollectFilesForCleaningUp(pyblish.api.InstancePlugin,
+                                AYONPyblishPluginMixin):
+    """Collect Files For Cleaning Up.
+    
+    This collector collects output files
+    and adds them to file remove list.
+
+    CAUTION:
+        This collector deletes the exported files and
+          deletes the parent folder if it was empty.
+        Artists are free to change the file path in the ROP node.
+    """
+
+    order = pyblish.api.CollectorOrder + 0.2  # it should run after CollectFrames
+
+    hosts = ["houdini"]
+    families = [
+        "camera",
+        "ass",
+        "pointcache",
+        "imagesequence",
+        "mantraifd",
+        "redshiftproxy",
+        "review",
+        "staticMesh",
+        "usd",
+        "vdbcache"
+    ]
+    label = "Collect Files For Cleaning Up"
+
+    def process(self, instance):
+
+        import hou
+
+        node = hou.node(instance.data["instance_node"])
+        output_parm = lib.get_output_parameter(node)
+        if not output_parm:
+            self.log.debug("ROP node type '{}' is not supported for cleaning up."
+                           .format(node.type().name()))
+            return
+        
+        filepath = output_parm.eval()
+        if not filepath:
+            self.log.warning("No filepath value to collect.")
+            return
+
+        files = []
+        # Non Render Products with frames
+        frames = instance.data.get("frames", [])
+        staging_dir, _ = os.path.split(filepath)
+        if isinstance(frames, str):
+            files = ["{}/{}".format(staging_dir, frames)]
+        else:
+            files = ["{}/{}".format(staging_dir, f) for f in frames]
+
+        # Render Products
+        expectedFiles = instance.data.get("expectedFiles", [])
+        for aov in expectedFiles:
+            for v in aov.values():
+                files += v
```

Comment:
Would this work?
```suggestion
        for aov in expectedFiles:
            files.extend(aov.values())
```

Reply:
I'll try and let you know.

Reply:
This worked. 
```python 
        for aov in expectedFiles:
            for v in aov.values():
                files.extend(v)
```
or in one line: 
```python
        for aov in expectedFiles:
            files.extend(sum(aov.values(), []))
```

Here's an example `expectedFiles`
```
expectedFiles = [
    {
    
        '_': ['//storage/work/ayon_projects/Robo/Assets/Character/warrior_robo/work/cfx/ayon/redshift_ropMain/render/redshift_ropMain.beauty.1001.exr', '//storage/work/ayon_projects/Robo/Assets/Character/warrior_robo/work/cfx/ayon/redshift_ropMain/render/redshift_ropMain.beauty.1002.exr'], 
        'Z': ['//storage/work/ayon_projects/Robo/Assets/Character/warrior_robo/work/cfx/ayon/redshift_ropMain/render/redshift_ropMain.Z.1001.exr', '//storage/work/ayon_projects/Robo/Assets/Character/warrior_robo/work/cfx/ayon/redshift_ropMain/render/redshift_ropMain.Z.1002.exr'], 
        'N': ['//storage/work/ayon_projects/Robo/Assets/Character/warrior_robo/work/cfx/ayon/redshift_ropMain/render/redshift_ropMain.N.1001.exr', '//storage/work/ayon_projects/Robo/Assets/Character/warrior_robo/work/cfx/ayon/redshift_ropMain/render/redshift_ropMain.N.1002.exr']
    }
]
```

Reply:
I used the one line version with a comment about it.